pkg/types/aws_resource_manager.go

```diff
@@ -67,6 +68,12 @@ type AWSResourceManager interface {
 	// GetAttributes operations but all we have (for new CRs at least) is a
 	// name for the resource
 	ARNFromName(string) string
+	// LateInitialize returns an AWS Resource after setting the late initialized
+	// fields from the readOne call. This method will initialize the optional fields
+	// which were not provided by the k8s user but were defaulted by the AWS service.
+	// If there are no such fields to be initialized, the returned object is similar to
```

Comment:
similar? or identical?

Also, is the implementation of `LateInitialize` responsible for setting a `ConditionTypeLateInitialized` condition on the resource? Or would a more robust solution be to have the reconciler handle setting a `ConditionTypeLateInitialized` condition on the resource and have the `LateInitialize` implementation only be responsible for returning *whether the resource was late initialized*?

Reply:
* identical. (PR updated)

> "is the implementation of LateInitialize responsible for setting a ConditionTypeLateInitialized condition on the resource?"

Yes, Currently this method encapsulates all the bits of LateInitialization from reconciler perspective. Reconciler is only responsible for patching resource and handling errors. (Updated description in the PR)